libs/platform/ebpf_object.c

```diff
@@ -196,6 +196,38 @@ ebpf_object_acquire_reference(_Inout_ ebpf_core_object_t* object)
     }
 }
 
+/**
+ * @brief Try to acquire a reference on an object. If the object is in the process of being freed, this function will
+ * fail.
+ *
+ * @param[in,out] object Object to acquire a reference on.
+ * @return true Reference was acquired.
+ * @return false Reference was not acquired.
+ */
+bool
+_ebpf_object_try_acquire_reference(_Inout_ ebpf_core_object_t* object)
```

Comment:
Is it possible to change this function to take pointer to `ebpf_base_object_t` instead of `ebpf_core_object_t`. Then this function can be directly reused by native module also, as `ebpf_native_module_t` also derives from `ebpf_base_object_t`.